elks/arch/i86/drivers/char/lp.c

```diff
@@ -32,219 +31,155 @@ static struct lp_info ports[LP_PORTS];
 
 #else
 
-/* extended to support some unusual ports; not used if BIOS_PORTS works */
-static struct lp_info ports[] = {
-    0x3BC, 0, 0,
-    0x378, 0, 0,
-    0x278, 0, 0,
-    0x260, 0, 0,
-    0x2E0, 0, 0,
-    0x2E8, 0, 0,
-    0x2F0, 0, 0,
-    0x2F8, 0, 0,
-    0x3E0, 0, 0,
-    0x3E8, 0, 0
+/* Probe the standard ports. */
+static struct lp_info ports[LP_PORTS] = {
+    {(char*)0x3BC, 0},
+    {(char*)0x378, 0},
+    {(char*)0x278, 0},
```

Comment:
I think this older use of casting to pointer was used because originally the argument of `outb_p` required it. Perhaps this should also be cleaned up to use `unsigned short io` in `struct lp_info` instead. I've cleaned up other parts of ELKS that also used pointer<->int casting. Another reason for this sort of thing throughout the older kernel was due to certain optimizations the old BCC compiler would perform on pointers, none of which is beneficial now.

Reply:
Interesting... I remember the warning that required the cast popped up later, and I didn't look closely at why.  Agreed that would be cleaner as `unsigned short io`.  I'll open a PR for that.  Thanks.